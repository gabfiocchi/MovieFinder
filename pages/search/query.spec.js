/* global jest describe, it, expect */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import ResultPage from './_query.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

const mockRoute = {
  query: {
    t: 'dummy'
  }
}
const mockRouterPush = jest.fn()
const mockDispatch = jest.fn()

const factory = () => {
  const store = new Vuex.Store({
    modules: {
      movies: {
        state: {
          items: [],
          isLoading: false
        },
        actions: {
          'movies/search': jest.fn()
        }
      }
    }
  })
  return shallowMount(ResultPage, {
    localVue,
    store,
    stubs: {
      fa: true
    },
    mocks: {
      $route: mockRoute,
      $router: {
        push: mockRouterPush
      }
    }
  })
}

describe.only('Results Page', () => {
  it('should render the component', () => {
    // Arrange.
    // Act.
    const wrapper = factory()
    // Assert.
    expect(wrapper).toMatchSnapshot()
  })

  it('resetSearch method', async () => {
    // Arrange.
    // Act.
    const wrapper = factory()
    await wrapper.find('form').trigger('click.prevent')
    // Assert.
    expect(mockRouterPush).toHaveBeenCalled()
  })

  it('fetch method', async () => {
    // Arrange.
    // Act.
    const wrapper = factory()
    wrapper.vm.$store.dispatch = mockDispatch
    await wrapper.vm.$options.fetch()
    // Assert.
    expect(mockDispatch).not.toHaveBeenCalled()
  })
})
