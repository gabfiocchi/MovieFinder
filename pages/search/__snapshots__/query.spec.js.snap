// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Page should render the component 1`] = `
<div class="container flex-col">
  <!---->
  <form class="max-w-md mx-auto px-6 py-3 border border-gray-200 border-opacity-75 rounded-lg shadow-lg w-full space-x-6 flex items-center bg-white"><button class="focus:outline-none">
      <fa-stub icon="search" class="text-gray-500"></fa-stub>
    </button> <input type="search" placeholder="Type some title..." class="w-full bg-transparent text-base focus:outline-none"> <button type="submit" class="bg-indigo-500 text-white py-2 text-sm px-3 rounded focus:outline-none">
      Search
    </button></form>
  <vue-horizontal-list-stub items="" options="[object Object]" class="w-full"></vue-horizontal-list-stub>
</div>
`;
